server/events/command_metrics_test.go

```diff
@@ -0,0 +1,148 @@
+package events
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/hashicorp/go-version"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/segmentio/stats"
+	"github.com/segmentio/stats/statstest"
+)
+
+type stepRunner struct{}
+
+func (c *stepRunner) Run(ctx models.ProjectCommandContext, extraArgs []string, path string) (string, error) {
+	return "success", nil
+}
+
+type errStepRunner struct{}
+
+func (c *errStepRunner) Run(ctx models.ProjectCommandContext, extraArgs []string, path string) (string, error) {
+	return "TLS handshake timeout", errors.New("error")
+}
+
+func TestInstrumentation(t *testing.T) {
+	tests := []struct {
+		scenario string
+		function func(*testing.T, *stats.Engine)
+	}{
+		{
+			scenario: "failing calls to ListRecentBuilds reports error metric",
+			function: testStepRunError,
+		},
+		{
+			scenario: "successful calls to Run reports timing and success metrics",
+			function: testStepRunSuccess,
+		},
+	}
+
+	for _, test := range tests {
+		testFunc := test.function
+		t.Run(test.scenario, func(t *testing.T) {
+			t.Parallel()
+			h := &statstest.Handler{}
+			testFunc(t, stats.NewEngine("test", h))
+		})
+	}
+}
+
+func testStepRunSuccess(t *testing.T, eng *stats.Engine) {
+	ic := InstrumentStepRunner(&stepRunner{}, eng, "plan")
+	tfVersion, _ := version.NewVersion("0.12.18")
+	projectContext := models.ProjectCommandContext{
+		Command:          models.PlanCommand,
+		Workspace:        "test-workspace",
+		TerraformVersion: tfVersion,
+	}
+	ic.Run(projectContext, nil, "/path")
+
+	found := measures(t, eng)
+	if len(found) != 2 {
+		t.Fatalf("expected 2 measures got %d", len(found))
+	}
+
+	checkHistogramEqual(t, found[0], stats.Measure{
+		Name:   "test.atlantis.steps.duration",
+		Fields: []stats.Field{stats.MakeField("", 1, stats.Histogram)},
```

Comment:
I'm not sure why this shows up as empty 🤷 @eculver

Reply:
I think it could be because the prefix hasn't been applied? Not sure though.